app/src/main/java/pl/gittobefit/WorkoutDisplay/objects/UserTrainings.java

```diff
@@ -26,8 +26,9 @@ static public UserTrainings getInstance()
         return INSTANCE;
     }
 
-    public void add(Training training)
+    public void add(Training training, int id)
     {
+        training.setId(id);
```

Comment:
Wydaje mi się, że lepiej tutaj nie ustawiać niczego, ponieważ sama metoda nie mówi o tym, że to robi. Chyba będzie lepiej, jakbyś przed tym ustawiał te id w treningu.